### ISSUE TITLE
Change remaining misuse of Ticket terminology

### ISSUE THREAD

**Issue by rgal** [SNSystems/llvm-project-prepo#8]
Ticket or Ticket File should now just refer to the stub files created in lieu of object files but in some places still refers to Digests.

RepoTicketGeneration should be RepoMetadataGeneration
RepoTicketNode should be RepoCompilationMember

[Tt]icket is also used incorrectly in some file names:
RepoTicket.cpp =>RepoDigest.cpp (or RepoCompilationMember.cpp)
RepoTicket.h =>RepoDigest.h (or RepoCompilationMember.h)
RepoTicketGeneration.cpp => RepoMetadataGeneration.cpp
and some tests which will need discerning which they are talking about.

Probably need some updates to the wiki too.

*note names currently under discussion*

**Comment by MaggieYingYi** [comment]
Following Paul’s advice (the PR#88 https://github.com/SNSystems/llvm-project-prepo/pull/88#issuecomment-621691193): “It would be great to record that discussion in a github issue”. I am putting the discussion here.

Maggie (from email):
```
1. Change the file name: 
    RepoTicket.cpp => RepoCompilationMember.cpp
    RepoTicket.h   => RepoCompilationMember.h                                                          
2. Inside these two files: 
   namespace name: ticketmd => compilationmembermd
   class name: TicketNode => CompilationMember
```

Paul (from email):
```
I’m not a fan of the “compilation member” term. The records inside a compilation are another thing that have been a challenge to name well. They’re similar to the “symbols” inside a traditional object file but there’s no equivalent of an “undefined” symbol. The best I’ve managed is “definition” since they always represent definitions that associate a name with a fragment.
```

Maggie (from email):
```
Could we use `RepoSymbol `? which is a repo symbol as you described above. It is different from the traditional symbols.
```

Paul (from email):
```
I’m wary of the term “symbol” because it’s used for a slightly different purpose in traditional object files
```

Russ (from email):
```
Maybe they are "symbols" then, just "Repo symbols" rather than ELF or object file symbols. I guess you're trying to avoid the baggage that comes with that name, but maybe it's easier to say "it's like this with differences" than "it's a whole new thing but a bit like that old thing...".
```

Paul (from email):
```
Yes, that’s exactly it. I’m wary of reusing terms from traditional object file (perhaps “TOF” from now on) because readers often — understandably — forget the differences from the thing with which they’re most familiar. I’m happier re-purposing terms from other domains because the differences tend to be less subtle and the audience less trained to a particular definition. I’ve been previously caught out where a term had a slightly different meaning than in conventional usage. The most dramatic example being the name of the project itself: originally “program database”, it had to be changed because people (in SN) assumed that it was the same as Microsoft’s PDB even after hearing a description of it that was clearly very, very different!

In a TOF the entries in a symbol table are either definitions or references. In ELF the section index associated with each symbol-table entry tells you which of the two it is. In contrast, a repo compilation has an array each entry of which represents a definition. The repo doesn’t need references to appear here because external fixups can simply name their target directly.

Perhaps my concerns are unwarranted: if everyone is happy with “symbol”, then let’s use it!
```

Russ (from email):
```
Or maybe just a "Record"/"Repo Record". It sounds a little tautologous to me but if it's the fundamental unit of operation in the program repository then I think that's okay.
```

Paul (from email):
```
To me as well. I don’t think that it’s quite fundamental enough to justify a name like that. We used to use “record” in pstore to denote a range of bytes but swapped that to “extent” (repurposing the term from file systems even though there are other definitions).
```

Maggie (from email):
```
After reading your concerns, I think we'd better not use the 'symbols'/'RepoSymbols'.
I am happy to change the file names to RepoDefinition.cpp and RepoDefinition.h.
```

Paul (from email):
```
Okay… let’s use “definition” instead of “compilation member”. Any dissenting voices?
```

Russ (from email):
```
Sounds okay to me.
```

Maggie (from email):
```
Should we change ticketmd to definitionmd, and TicketNode to Definition?
``` 

Paul (from email):
```
TicketNode (assuming that this is the name that appears in IR files, perhaps the previous one as well) should really be prefixed with “Repo” to make it clear to anyone unfamiliar that this is repo-specific metadata. Things inside RepoDefinition.cpp [etc.] don’t need the prefix since they inherit context from their location.
```

Maggie (from email):
```
Actually, both `ticketmd` (namespace) and `TicketNode` (class name) are defined in the RepoDefinition.cpp File. However, I think they still need the prefix since they are used outside of file. For example, in the IR code or the RepoMetadataGenerationPass.cpp. With the prefixed, make the code clearer. Changed:
ticketmd => repodefinitionmd
TicketNode => RepoDefinition
```

Paul (from email):
```
FTR, you wouldn’t need the prefix in RepoMetadataGenerationPass.cpp (because it’s already there in the filename). Obviously, we need it in IR code.
Is there something in the LLVM style guide that mandates a namespace name like “repodefinitionmd”?
```

Maggie (from email):
```
I can't find name style of namespace in the style guide page, but I checked the LLVM code. The following two examples:
namespace cycleclock which is defined in the cycleclock.h header file, and namespace codeview which is defined in the CodeView.h file.
Maybe change "repodefinationmd" to "repodefinition" (Similar as above two examples: namespace name is the same as the file name)?
```

Paul (from email):
```
Yes, I think that’s a good idea.
```

**Comment by MaggieYingYi** [comment]
Fixed in the commit 37f1302.
